core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/Region.scala

```diff
@@ -1,26 +1,23 @@
 package edu.uci.ics.amber.engine.architecture.scheduling
 
-import edu.uci.ics.amber.core.workflow.PhysicalOp
-import edu.uci.ics.amber.engine.architecture.scheduling.config.ResourceConfig
 import edu.uci.ics.amber.core.virtualidentity.PhysicalOpIdentity
-import edu.uci.ics.amber.core.workflow.{PhysicalLink, PortIdentity}
+import edu.uci.ics.amber.core.workflow.{GlobalPortIdentity, PhysicalLink, PhysicalOp}
+import edu.uci.ics.amber.engine.architecture.scheduling.config.ResourceConfig
 import org.jgrapht.graph.{DefaultEdge, DirectedAcyclicGraph}
 import org.jgrapht.traverse.TopologicalOrderIterator
 
+import java.net.URI
 import scala.jdk.CollectionConverters.IteratorHasAsScala
 
 case class RegionLink(fromRegionId: RegionIdentity, toRegionId: RegionIdentity)
 
 case class RegionIdentity(id: Long)
-
-case class GlobalPortIdentity(opId: PhysicalOpIdentity, portId: PortIdentity, input: Boolean)
-
 case class Region(
     id: RegionIdentity,
     physicalOps: Set[PhysicalOp],
     physicalLinks: Set[PhysicalLink],
     resourceConfig: Option[ResourceConfig] = None,
-    materializedPortIds: Set[GlobalPortIdentity] = Set.empty
+    storageURIs: Map[GlobalPortIdentity, URI] = Map.empty
```

Comment:
sorry I just realized this as a major comment:

per our discussion before, could you please put this `storageURI` into the `resourceConfig`? The reason is that storage is a kind of resource, to be decided by the scheduler. The `resourceConfig` will be used by cost estimator to determin the cost of a region, and added up the be the cost of a schedule, and the number of storages (determined by the number of storage URIs) is good information for cost. 

Down the road we may take `resourceConfig` outside `Region` class to make `Region` immutable.

Reply:
Moved `storageURIs` to `resourceConfig`.